scikeras/utils/__init__.py

```diff
@@ -12,6 +12,19 @@
 from tensorflow.keras.metrics import get as get_metric
 from tensorflow.keras.metrics import serialize as serialize_metric
 
+from scikeras.utils._transformers import (
+    BaseKerasClassifierFeatureTransformer,
+    BaseKerasClassifierTargetTransformer,
+    BaseKerasRegressorFeatureTransformer,
+    BaseKerasRegressorTargetTransformer,
+    BaseScikerasDataTransformer,
+    Ensure2DTransformer,
+    KerasClassifierFeatureTransformer,
+    KerasClassifierTargetTransformer,
+    KerasRegressorFeatureTransformer,
+    KerasRegressorTargetTransformer,
```

Comment:
The goal here was to establish an "interface" for each transformer. Sadly, Python does not supports interfaces as intended here, so I settled on abstract classes.

`BaseScikerasDataTransformer` certainly achieves this because it forces implementation of the `get_metadata` method. 

`BaseKerasClassifierTargetTransformer` a bit less so because it cannot force signature compatibility (what we really want is to force the `return_proba` parameter).

`BaseKerasRegressorFeatureTransformer ` I agree is pretty useless, it's only there because it felt wrong to have a `BaseKerasClassifierTargetTransformer` without a corresponding interface/base class for KerasRegressor.

That said, do you still think we should keep only the 4 that you mentioned?

Reply:
Regarding the namespace, no problem, it'd be easy enough to delete the imports here and rename `_transformers.py` to `transformers.py`.